src/Utility/Hash.php

```diff
@@ -267,15 +267,17 @@ protected static function _matches($data, $selector)
                 $prop = $prop ? '1' : '0';
             } elseif ($isBool) {
                 $prop = $prop ? 'true' : 'false';
+            } elseif (is_numeric($prop)) {
+                $prop = (string)$prop;
```

Comment:
How will integer data be matched now?

Reply:
Ex:
```php
$data = [
            ['value' => '1'],
            ['value' => 1],
            ['value' => 'string-value'],
        ];
$result = Hash::extract($data, '{n}[value=1]');
```
then
```php
$result = [
    ['value' => '3'],
    ['value' => 3],
]
```

---
When `$selector` is `'[value=3]'`, `$val` is given as string.
And `$prop` casting to string, so `$prop === $val` means `'3' === '3'`, returns true.